### ISSUE TITLE
FRASER

### ISSUE THREAD

**Issue by ischeller** [Bioconductor/Contributions#1320]
Update the following URL to point to the GitHub repository of
the package you wish to submit to _Bioconductor_

- Repository: https://github.com/gagneurlab/FRASER

Confirm the following by editing each check box to '[x]'

- [x] I understand that by submitting my package to _Bioconductor_,
  the package source and all review commentary are visible to the
  general public.

- [x] I have read the _Bioconductor_ [Package Submission][2]
  instructions. My package is consistent with the _Bioconductor_
  [Package Guidelines][1].

- [x] I understand that a minimum requirement for package acceptance 
  is to pass R CMD check and R CMD BiocCheck with no ERROR or WARNINGS. 
  Passing these checks does not result in automatic acceptance. The 
  package will then undergo a formal review and recommendations for 
  acceptance regarding other Bioconductor standards will be addressed.

- [x] My package addresses statistical or bioinformatic issues related
  to the analysis and comprehension of high throughput genomic data.

- [x] I am committed to the long-term maintenance of my package. This
  includes monitoring the [support site][3] for issues that users may
  have, subscribing to the [bioc-devel][4] mailing list to stay aware
  of developments in the _Bioconductor_ community, responding promptly
  to requests for updates from the Core team in response to changes in
  _R_ or underlying software.

I am familiar with the essential aspects of _Bioconductor_ software
management, including:

- [x] The 'devel' branch for new packages and features.
- [x] The stable 'release' branch, made available every six
      months, for bug fixes.
- [x] _Bioconductor_ version control using [Git][5]
  (optionally [via GitHub][6]).

For help with submitting your package, please subscribe and post questions
to the [bioc-devel][4] mailing list.

[1]: https://bioconductor.org/developers/package-guidelines/
[2]: https://bioconductor.org/developers/package-submission/
[3]: https://support.bioconductor.org
[4]: https://stat.ethz.ch/mailman/listinfo/bioc-devel
[5]: http://bioconductor.org/developers/how-to/git/
[6]: http://bioconductor.org/developers/how-to/git/sync-existing-repositories/

**Comment by lshep** [comment]
Are you using a dataset?  Perhaps a smaller dataset would speed things along?

**Comment by lshep** [comment]
If you haven't corrected it already - it looks like your package was/is  writing to a users home directory which is unacceptable.  Please write to a `tempdir( )`  or `tempfile( ) `.

**Comment by c-mertes** [comment]
Dear @lshep 
thanks already for your comments. We still try to pass all the BiocChecks (including not writing to the home folder). 
Currently we are still failing on Windows on building the vignette. The error looks like a configuration problem. Or do we have to set up the vignette builder differently? 

```
Error: processing vignette 'FraseR.Rnw' failed with diagnostics:
Failed to locate 'texi2pdf' output file 'FraseR.pdf' for vignette with name 'FraseR' and engine 'knitr::knitr'.
```

We do include a png in the vignette, but thats all. Everything else is plain R code and text.
Any suggestions are welcome?

**Comment by lshep** [comment]
I'll try and look into this more in the next few days  -  we were experiencing some knitr issues (that we have reported directly to the knitr maintainer) that may or may not be part of the issue.

**Comment by lshep** [comment]
I'm still not quite sure what is causing the ERROR on windows but we don't see this in other Bioconductor packages so I'm inclined to believe its not on our end.  I'll try to look more into it but please also do the same. 

Some clarifications questions and comments below: 

- [ ] The check time is really long? Are you only using minimal datasets for examples/vignettes/tests? Do you have an idea of where/what is causing the long check time?  It looks like you are using 12 samples for example data can this be truncated? 

-  [ ] You include bam/bai files with the package. Can existing bam/bai files that are already in Biocondcutor be used?  There are some in the ExperimentHub
```
library(ExperimentHub)
hub = ExperimentHub()
mcols(query(hub, "bai"))
```
There are also some in a number of existing Bioconductor packages - most notably Rsamtools that you already import. I did a quick search and there are 66 packages already in Bioconductor that provide bam/bai files in their extdata that could potentially be utilized? If you need to keep your own please explain why. 

- [ ] Any data that remains in inst/extdata we would require an accompanied inst/script directory that has source and curation information.  Where did data come from and how it was generated.  It could be text, sudo-code, or code but should indicate how each of the files in inst/extdata came to be. 

- [ ] You export a lot of functions - Are all these functions intended to be accessible/usable by the end user? 

- [ ] The R check seems to indicate the R/doc/libs directories are of sizable note? 

- [ ] Are the codecov.yml, appveyor.yml and setupEnv.R in the top directory related to running on travis? Can these be .gitignored?

**Vignette inspired questions**

- [X] This is a very impressive packages based on the very detailed Vignette. We appreciate you took the time to document well. 

- [ ] There is a section in a coding block that is #skipped here because not meaniful on the small example but maybe at least comment on what it is suppose to do or reference the detailed section below. 

- [ ] It seems throughout the vignette the references are not displaying.  All appear to be [?] or [??] - although I think the [??] was a placeholder? 
 
- [ ] 3.2 Data preprocessing and QC - please refer to the gene expression slides? Was there suppose to be a link or some other documentation? 

- [ ] 3.3.2 You reference using biomart to annotation but in this case isn't this not correct? You used the precompiled txdb and org.db packages? 

I thought I would post these comments first since there is a lot of R code for me to look through. That way you could get started with some responses.  I will provide any feedback on any coding issues at a later time.

**Comment by ischeller** [comment]
Dear @lshep ,
Thank you for already providing feedback to us. We are currently going through your suggestions and are working on fixing the issues that you spottted.

**Comment by lshep** [comment]
is this ready for a re-review?  could you summarize the changes?

**Comment by lshep** [comment]
Is this ready for a re-review?  If so please bump the version to get a new report and summarize the changes here.

**Comment by c-mertes** [comment]
Dear @lshep 
we will polish it up over the next view days and let you know when it is ready for review. 
Our current problems are 

1. to reduce testing runtime to meet the 10 minutes requirement
1. `! LaTeX Error: File `geometry.sty' not found.` on tokay2. 

The first one we try to meet in the next few days, but the second one, I think, has to be fixed on bioc team side.

**Comment by lshep** [comment]
Thanks.  Let me know when its ready for review.  If possible try to use the minimum dataset for examples which is often good enough to show functionality rather than a larger example.   I will look into the latex error on windows but for now ignore.

**Comment by lshep** [comment]
Please remember that all packages need to be reviewed and accepted by April 22 to be included in the 3.11 release,  else they will be released in the fall 3.12 release.

**Comment by ischeller** [comment]
Dear @lshep, we finally pushed our package FRASER to be ready for a review. 

We tried to manage to beat the `R CMD check` 10 minutes timeline, but did not succeed  unfortunately. We cut already down a lot of examples to a minimum and also reduced the tests. But we would like to keep this as otherwise we would reduce the quality of the package in our point of view. 

Here is our reply to your comment: https://github.com/Bioconductor/Contributions/issues/1320#issuecomment-579311282_

* We reduced the number of samples to speed up the computation. 
* We included bam files since we do want to look at rare splicing events. Currently all bam files I saw in the Experiment hub are from alternative splicing not including novel splicing events. Hence we would like to keep our 3 bam files within the package (660kb in total). 
  * we added the data curation information. 
* The most functions we export are accessor function and plotting function. We think that most of the functions make it easy for the end user. But you can let us know after the review if you think we should not export some of them. 
* We have some C++ code for our gradients of the autoencoder. Here I'm not sure how to reduce this.
* We do run on `travis` and on `appveyor` CI jobs to make sure we have a working package. Hence we would like to keep them at least in our private git repository. We can later if required delete them on the Bioconductor repository. 

**Vignette**
* We now use the function as intended by the workflow also on the small dataset. 
* Now the references should work (at least on our side they do)
* We added the link to our ASHG19 workshop. 
* We rephrased the paragraph in 3.3.2 to reflect the two options: `biomaRt` and `txdb`

**Comment by lshep** [comment]
Overall I think the code is really well written.  Thank you for making the adjustments.  A few minor comments: 

**README**

- [ ] Add Bioconductor installation instructions to the readme

**Vignette**

- [ ] it doesn't look like the plotVolcano plot is rendering right? This could
  only be an issue for me but wanted you to be aware of it 
```
> plotVolcano(fds, sampleID="sample1", type="psi5", aggregate=TRUE)
Warning message:
In grid.Call.graphics(C_points, x$x, x$y, x$pch, x$size) :
  semi-transparency is not supported on this device: reported only once per page

```

- [ ] Can the argument junctions and spliceSites take GRange objects directly?
  This should be really easy to implement if you just check the class of the
  object being brought in and then convert to whatever format you need? 

There is a lot of code but it seems generally well organized and well documented.

**Comment by c-mertes** [comment]
Dear @lshep,
thanks for the review. 
* I added the installation instructions to the readme.
* `FraserDataSet` already was able to take `GRange` as input. I added it to the documentation to make it clear.
* To make it coherent with our other package [`OUTRIDER`](http://bioconductor.org/packages/release/bioc/html/OUTRIDER.html) we changes FraseR to FRASER and hence FraserDataSet.

* Thanks for the `warning` report. I never saw this warning, but I guess this is due to the `Cairo` package [see here](https://tinyheero.github.io/2015/09/15/semi-transparency-r.html). We are not sure if we find a fix right away, but will look into it. What OS are you using MacOS?

Let us know if you need more changes.

**Comment by lshep** [comment]
I'm using a linux (but very outdated ubuntu 14.04 -- my updated laptop is delayed from quarantine)  so it might be my own issue.  We can proceed and just be aware of the potential issue if others start to inquire. I think everything looks good and will accept.

**Comment by c-mertes** [comment]
@lshep thanks a lot for the help and for the push before the release!

**Comment by mtmorgan** [comment]
The master branch of your GitHub repository has been added to Bioconductor's git repository.

To use the git.bioconductor.org repository, we need an 'ssh' key to associate with your github user name. If your GitHub account already has ssh public keys (https://github.com/ischeller.keys is not empty), then no further steps are required. Otherwise, do the following:

1. [Add an SSH key to your github account][]
1. [Submit your SSH key to Bioconductor][]

See further instructions at

https://bioconductor.org/developers/how-to/git/

for working with this repository. See especially

https://bioconductor.org/developers/how-to/git/new-package-workflow/
https://bioconductor.org/developers/how-to/git/sync-existing-repositories/

to keep your GitHub and Bioconductor repositories in sync.

Your package will be included in the next nigthly 'devel' build (check-out from git at about 6 pm Eastern; build completion around 2pm Eastern the next day) at

https://bioconductor.org/checkResults/

(Builds sometimes fail, so ensure that the date stamps on the main landing page are consistent with the addition of your package). Once the package builds successfully, you package will be available for download in the 'Devel' version of Bioconductor using `BiocManager::install("FRASER")`. The package 'landing page' will be created at

https://bioconductor.org/packages/FRASER

If you have any questions, please contact the bioc-devel mailing list (https://stat.ethz.ch/mailman/listinfo/bioc-devel); this issue will not be monitored further.

[Add an SSH key to your github account]: https://help.github.com/articles/connecting-to-github-with-ssh/
[Submit your SSH key to Bioconductor]: https://git.bioconductor.org/BiocCredentials/
